test/regressions/index.js

```diff
@@ -0,0 +1,209 @@
+  // /^docs-guides-.*/, // No public components
+];
+
+const unusedBlacklistPatterns = new Set(blacklist);
+
+function excludeTest(suite, name) {
+  return blacklist.some((pattern) => {
+    if (typeof pattern === 'string') {
+      if (pattern === suite) {
+        unusedBlacklistPatterns.delete(pattern);
+
+        return true;
+      }
+      if (pattern === `${suite}/${name}.png`) {
+        unusedBlacklistPatterns.delete(pattern);
+
+        return true;
+      }
+
+      return false;
+    }
+
+    // assume regex
+    if (pattern.test(suite)) {
+      unusedBlacklistPatterns.delete(pattern);
+      return true;
+    }
+    return false;
+  });
+}
+
+// Get all the tests specifically written for preventing regressions.
+const requireStories = require.context('packages/storybook/src/stories', true, /\.(js|ts|tsx)$/);
+const stories = requireStories.keys().reduce((res, path) => {
+  let suite = path
+    .replace('./', '')
+    .replace('.stories', '')
+    .replace(/\.\w+$/, '');
+  suite = `stories-${suite}`;
+
+  const cases = requireStories(path);
+
+  Object.keys(cases).forEach((name) => {
+    if (name !== 'default' && !excludeTest(suite, name)) {
+      res.push({
+        path,
+        suite,
+        name,
+        case: cases[name],
+      });
+    }
+  });
+
+  return res;
+}, []);
+
+// Also use some of the demos to avoid code duplication.
+const requireDocs = require.context('docsx/src/pages', true, /js$/);
+const docs = requireDocs.keys().reduce((res, path) => {
+  const [name, ...suiteArray] = path.replace('./', '').replace('.js', '').split('/').reverse();
+  const suite = `docs-${suiteArray.reverse().join('-')}`;
+
+  if (excludeTest(suite, name)) {
+    return res;
+  }
+
+  res.push({
+    path,
+    suite,
+    name,
+    case: requireDocs(path).default,
+  });
+
+  return res;
+}, []);
+
+const tests = stories.concat(docs);
+
+if (unusedBlacklistPatterns.size > 0) {
+  console.warn(
+    `The following patterns are unused:\n\n${Array.from(unusedBlacklistPatterns)
+      .map((pattern) => `- ${pattern}`)
+      .join('\n')}`,
+  );
+}
+
+function App() {
+  function computeIsDev() {
+    if (window.location.hash === '#dev') {
+      return true;
+    }
+    if (window.location.hash === '#no-dev') {
+      return false;
+    }
+    return process.env.NODE_ENV === 'development';
+  }
+  const [isDev, setDev] = React.useState(computeIsDev);
+  React.useEffect(() => {
+    function handleHashChange() {
+      setDev(computeIsDev());
+    }
+    window.addEventListener('hashchange', handleHashChange);
+
+    return () => {
+      window.removeEventListener('hashchange', handleHashChange);
+    };
+  }, []);
+
+  // Using <link rel="stylesheet" /> does not apply the google Roboto font in chromium headless/headfull.
+  const [fontState, setFontState] = React.useState('pending');
+  React.useEffect(() => {
+    webfontloader.load({
+      google: {
+        families: ['Roboto:300,400,500,700', 'Material+Icons'],
```

Comment:
We don't use icon font

```suggestion
        families: ['Roboto:300,400,500,700'],
```